### PR TITLE
[4.0] PHP 8.1 deprecation notices various

### PR DESCRIPTION
**Pull request by frostmakk**
Pull Request for Issue # .

### Summary of Changes
Patching PHP 8.1 deprecation notices where null is provided instead of 'something'.

### Testing Instructions
Code review or checking logs.

### Actual result BEFORE applying this Pull Request
Deprecation notices

### Expected result AFTER applying this Pull Request
No deprecation notices from these files.

### Documentation Changes Required
Probably not.

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 318edec93096f755c8e6a10fffa08d9ad12f3e7e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36362">issues.joomla.org/tracker/joomla-cms/36362</a>.</sub>

File: layouts/joomla/form/field/url.php [hunk 1]
```diff
@@ -69,5 +69,5 @@
 	name="<?php echo $name; ?>"
 	<?php echo !empty($class) ? ' class="form-control ' . $class . '"' : 'class="form-control"'; ?>
 	id="<?php echo $id; ?>"
-	value="<?php echo htmlspecialchars(PunycodeHelper::urlToUTF8($value), ENT_COMPAT, 'UTF-8'); ?>"
+	value="<?php echo htmlspecialchars(PunycodeHelper::urlToUTF8($value) ?? '', ENT_COMPAT, 'UTF-8'); ?>"
```
Comment (HLeithner) [inline layouts/joomla/form/field/url.php]: please revert this here and fix it in https://github.com/joomla/joomla-cms/blob/318edec93096f755c8e6a10fffa08d9ad12f3e7e/libraries/src/String/PunycodeHelper.php#L135

**Comment by frostmakk** [comment]
@HLeithner  Sorry. I put the PunicodeHelper in the library PR instead of here.
Is it the newly merged system tests that make this fail?

**Comment by frostmakk** [comment]
This looks a bit suspicious in the api-test  
Undeclared variable?

image

**Comment by HLeithner** [comment]
Can you update tge Branch please

**Comment by frostmakk** [comment]
Sorry. I don't understand.

**Comment by HLeithner** [comment]
merge the current 4.0-dev branch with this pr branch

**Comment by HLeithner** [comment]
I have tested this item :white_check_mark: successfully on c93dd560b0c569ca8eeb6c3542e3fadc923dfefa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36362">issues.joomla.org/tracker/joomla-cms/36362</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**